usr/share/rear/output/USB/Linux-i386/300_create_grub.sh

```diff
@@ -0,0 +1,64 @@
+# only run for grub
+if [ -z $USB_BOOTLOADER ] && [[ ! $USB_BOOTLOADER =~ grub ]]; then
+    return 0
```

Comment:
I do not undrestand the condition. I think
```
if [ -z $USB_BOOTLOADER ] && [[ ! $USB_BOOTLOADER =~ grub ]]
```
means
```
if $USB_BOOTLOADER is empty AND $USB_BOOTLOADER is not "grub"
```
but I assume output/USB/Linux-i386/300_create_grub.s should be skipped
```
if $USB_BOOTLOADER is empty OR $USB_BOOTLOADER is not "grub"
```
e.g.
```
# USB_BOOTLOADER='' ; if [ -z $USB_BOOTLOADER ] && [[ ! $USB_BOOTLOADER =~ grub ]] ; then echo no grub ; else echo grub ; fi
no grub

# USB_BOOTLOADER='grub' ; if [ -z $USB_BOOTLOADER ] && [[ ! $USB_BOOTLOADER =~ grub ]] ; then echo no grub ; else echo grub ; fi
grub

# USB_BOOTLOADER='foo' ; if [ -z $USB_BOOTLOADER ] && [[ ! $USB_BOOTLOADER =~ grub ]] ; then echo no grub ; else echo grub ; fi
grub
```
I think the test should be simply
```
if $USB_BOOTLOADER is not "grub"
```
i.e.
```
# Skip when USB_BOOTLOADER is not 'grub':
test "$USB_BOOTLOADER" = "grub" || return 0
```

Reply:
the empty is in there for empty beeing previous behaviour as described in the conf file comment.
the -z is very common and readable bash to me for empty checks.
the not equal check van be written differently of course.